### PR TITLE
[Answer Query Using Materialized Views] Refactor codes to new file aqumv.c

### PR DESCRIPTION
**Pull request by avamingli**
Refactor to external func **answer_query_using_materialized_views()** with some refinements.
Rename GUC to **enable_answer_query_using_materialized_views** to avoid duplicated name with implementation function.
New function **aqumv_init_context()** to init context of rewrite.

Authored-by: Zhang Mingli avamingli@gmail.com

fix #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

File: src/backend/optimizer/plan/aqumv.c [hunk 1]
```diff
@@ -0,0 +1,796 @@
...
+		return mv_final_rel;
+
+	/*
+	 * AQUMV_FIXME_MVP:
+	 *	Single relation, excluding catalog/inherit/partition tables.
+	 */
+	if (list_length(parse->jointree->fromlist) != 1)
+		return mv_final_rel;
+
+    jtnode = (Node *) linitial(parse->jointree->fromlist);
```
Comment (yjhjstz) [inline src/backend/optimizer/plan/aqumv.c:136]: align code.

File: src/backend/optimizer/plan/aqumv.c [hunk 2]
```diff
@@ -0,0 +1,796 @@
...
+
+		/*
+		 * AQUMV
+		 * Build a plan of new SQL.
+		 * AQUMV is cost-based, let planner decide which is better.
+		 * AQUMV_FIXME_MVP:
+		 * 	no qp_callback function now.
+		 * 	replcace one-by-one?
+		 */
+		mv_final_rel = query_planner(subroot, NULL, NULL);
```
Comment (yjhjstz) [inline src/backend/optimizer/plan/aqumv.c:401]: why no callback ?

File: src/backend/optimizer/plan/aqumv.c (hunk 1 above)
Comment (avamingli) [reply src/backend/optimizer/plan/aqumv.c]: Fixed.

File: src/backend/optimizer/plan/aqumv.c (hunk 2 above)
Comment (avamingli) [reply src/backend/optimizer/plan/aqumv.c:401]: Hi, thanks for review.
AQUMV don't  support window/group/distinct clause now. We don't need a callback function to compute pathkeys for them.

**Review by yjhjstz — APPROVED** [review]
LGTM

**Pull request merged**